google-cloud-spanner/src/main/java/com/google/cloud/spanner/AbstractResultSet.java

```diff
@@ -786,7 +786,7 @@ protected Float64Array getDoubleListInternal(int columnIndex) {
     private final Statement statement;
 
     private SpannerRpc.StreamingCall call;
-    private boolean withBeginTransaction;
+    private volatile boolean withBeginTransaction;
```

Comment:
This value is read by multiple threads (in the tests) and could therefore sometimes return a stale value without the `volatile` keyword.